Simplify API and add some QOL

ZanzyTHEbar opened the pull request:
> `QOL` features: 
- eslint
- .editorconfig
- vscode workspace

> `API` Changes
i simplified the `typedCharts.tsx` file to export directly so that users can simply import the chart that they want: 
```tsx
import { Line } from 'solid-chartjs'
```
I also simplified the api so that a charts required registerables are set up for them. Just import a chart and use it :) 
I renamed the `ChartJs` component to `DefaultChart` and have left it un-registered - for the most flexibility. 

I can setup `semantic-release` for versioning as well, and suggest using the github actions to publish to npm.

ZanzyTHEbar committed:
feat initial commit

ZanzyTHEbar committed:
fix: temporary fix for actions

- disabled TypeChecking for now

ZanzyTHEbar committed:
Format

ZanzyTHEbar committed:
fix: clean up typedcharts and readme

- simplified api
- updated readme to reflect api
- add dev to workspace folder

ZanzyTHEbar committed:
refactor: remove hello world component

ZanzyTHEbar committed:
fix: clean up typedcharts and readme

- simplified api even more
- added all chart required registerables by default

ZanzyTHEbar committed:
Format

ZanzyTHEbar committed:
docs: update readme to reflect new api

s0ftik3 commented:
is it a draft? I see a lot of not completed code parts. Some of them seems to be excess. Also, you've removed all the demo functional in /dev folder (Randomize, Add Dataset, Remove Dataset, and ability to change the chart's type and resize it). I think it'd be nice to have them stay. Either way, thanks for the work you've done! I will check the entire PR tomorrow. I think we can get it fully done and it will be merged.

ZanzyTHEbar commented:
> is it a draft? I see a lot of not completed code parts. Some of them seems to be excess. Also, you've removed all the demo functional in /dev folder (Randomize, Add Dataset, Remove Dataset, and ability to change the chart's type and resize it). I think it'd be nice to have them stay. Either way, thanks for the work you've done! I will check the entire PR tomorrow. I think we can get it fully done and it will be merged.

Yes, apologies, I am going to add back in all of the demo sections later. I had to stop for dinner - lol. On my to-do list 👍

ZanzyTHEbar committed:
fix: fix forward ref of chart

- cleaned up types
- added forward ref support

ZanzyTHEbar committed:
refactor: remove unused utils.ts

ZanzyTHEbar committed:
fix: add back in demo

- fix some type annotation errors
- added color to chart
- added ref as example
- changed line to Default chart in demo

ZanzyTHEbar commented:
Alright - cleaned everything up and added the demo back in :) should be good to go, lmk how you like it.

ZanzyTHEbar committed:
fix: make ref optional

ZanzyTHEbar committed:
fix: update tsup config

ZanzyTHEbar committed:
fix: tests failing due to server

ZanzyTHEbar committed:
fix: update package.json

ZanzyTHEbar committed:
fix: update package.json

ZanzyTHEbar committed:
Format

ZanzyTHEbar committed:
fix: update package.json

ZanzyTHEbar committed:
fix: update package.json

ZanzyTHEbar committed:
Format

s0ftik3 commented:
I did have pretty similar function yet I decided not to use it because it doesn't have friendly set of colors so I just preferred the default one.

https://github.com/ZanzyTHEbar/solid-chartjs/blob/dd67fc67affd85cfd02a7c9cca1fb6ed4f6f96e1/dev/utils.ts#L39

You didn't restore the styles in `App.module.css` so the demo page now looks like this:
<img width="1680" alt="image" src="https://github.com/s0ftik3/solid-chartjs/assets/64411317/c8331c84-d431-40f9-b57c-87f7030a04f8">
Instead of:
<img width="1670" alt="image" src="https://github.com/s0ftik3/solid-chartjs/assets/64411317/9c249e55-2888-4d17-9868-64761d5cd0ac">
 
 `AllCharts.tsx` is just a default SolidJs page with a bunch of charts in below. I think it'd be better to use the restored styles and remove all the SolidJs demo elements (such as logo `logo.svg`, the default page styles `styles.css`, etc.)
 
 I like the refactored `chart.tsx` version. By the way, have you tried to dynamically change the type of the chart? In this part of the code it's implemented:

https://github.com/ZanzyTHEbar/solid-chartjs/blob/dd67fc67affd85cfd02a7c9cca1fb6ed4f6f96e1/src/chart.tsx#L67

however, it doesn't work properly and I'm not sure it's a bug related to the SolidJs implementation or Chart.js itself (I've even created an issue where I described the problem more detailed — https://github.com/chartjs/Chart.js/issues/11291

Also, thank you very much for documenting the types of the lib!

And the last excess thing is the `test` folder which is also a part of initial commit. I guess we could either cover the lib with tests or just remove the folder. I think that if we have something that we don't use, we should consider to remove it. But I think even if you decide to remove it, I will still cover it with tests sooner or later :DD

Ah, and the last thing is `package.json`. Please, restore the previous version of `package.json`, specifically the package's version (1.2.4). Once I merge this PR, I will bump the version myself. Also, I see you used these links `https://github.com/solid-community/solid-chartjs.git` and `https://github.com/solid-community/solid-chartjs/issues` which lead to nowhere, so far.

Speaking of the rest, I think it's much better that it was. I guess I'll cover it with tests and improve the demo a little bit (chart colors and switching between App.tsx and AllCharts.tsx) and the lib will be pretty much done. Didn't think it will go that far, I haven't even installed and used it in my work project yet, kek.

Thanks again!

ZanzyTHEbar commented:
Interesting, yeah colors is personal preference. I think it's better to show the functionality in a library than to choose based on personal preference. 

I was able to replicate the same bug - not sure what causes it yet, but i am looking into it. Will subscribe to the issue you created. 

I'll add back in the CSS stylings :) - forgot about that, apologies.

Yeah, I considered removing the `test` folder - however i thought it was nice to leave it, as this library should be covered in tests. To remove the test folder would also require changing the YAML files. 

As far as versioning, i think that should be done with `semantic-release` and i feel that setting up automating versioning based on commit history and some commit rules, and setting up publishing to npm is the best option. I do that for my other repos, and it works very well. Provides some structure to versioning, and allows a nice level of automation to changes in the library. 

To that end, i will change the version back in the `package.json` - however i will also implement `semantic-release`, with a clear readme on how to use it (it's very easy to use). 

Haha - yeah the repo urls were placeholders, forgot to change those. I intended on trying to publish this into the solid-community like with the `solid-three` lib that i am helping to work on, but it's fine to just `pr` into the ecosystem instead. 

I have perms to the `solid-community` repo though, so if you feel like this lib would be better maintained by the community repo, we can discuss it with `davebased`. I am sure he'd be happy for us to do that. 

All in all, very happy that you like my changes :) looking forward to seeing this on npm! Finally, a _decent_ charting lib for solid.

s0ftik3 commented:
> Interesting, yeah colors is personal preference. I think it's better to show the functionality in a library than to choose based on personal preference.

true, yet I just wanted it to look like on chartjs.org website. You might've noticed that I stole their design xD

> Changing the chart type works perfectly fine for me, so not sure. Perhaps you should erase your browser cache? ctrl + F5 on windows. I've already implemented my fork into a personal project, and changing the chart type works there as well. Works find in the demo. I am not experiencing the same bug as you - perhaps it's your impl? have you tried this pull request/my fork?

Yeah, I've tried it on both your and mine repos. For some reason, I still could reproduce it. I'll look forward it later in the evening when I get free.

> Yeah, I considered removing the test folder - however i thought it was nice to leave it, as this library should be covered in tests. To remove the test folder would also require changing the YAML files.

Don't even bother then, leave it. I also thought it'd be nice to have tests. Also, it's not that big library to struggle covering it with tests, in my opinion. So, I think you and I may consider do it in the future.

> As far as versioning, i think that should be done with semantic-release and i feel that setting up automating versioning based on commit history and some commit rules, and setting up publishing to npm is the best option. I do that for my other repos, and it works very well. Provides some structure to versioning, and allows a nice level of automation to changes in the library.

I'll check it out, thanks! Never actually used it before.

> I have perms to the solid-community repo though, so if you feel like this lib would be better maintained by the community repo, we can discuss it with davebased. I am sure he'd be happy for us to do that.

Guess we'll see how it goes. I'll have a vacation soon, so I'll have much more time for side projects. I've had a really long pause in commits on Github for over the last few years.

Let me know once you're done. Also, I think we could just create a dev branch for both of us to push fixes or some new implementations. No need in forking, much closer to gitlab flow actually. lmk if you're good with it.

ZanzyTHEbar commented:
>  Let me know once you're done. Also, I think we could just create a dev branch for both of us to push fixes or some new implementations. No need in forking, much closer to gitlab flow actually. lmk if you're good with it.

Yeah - i am happy with that. Better than forking, would make changes easier and more streamlined. 

> Let me know once you're done. Also, I think we could just create a dev branch for both of us to push fixes or some new implementations. No need in forking, much closer to gitlab flow actually. lmk if you're good with it.

I just spun up my fork and tried to replicate the bug, i was able to. Apologies for my earlier statement. I am able to reliably reproduce the bug by going to the `polarArea` or `radar` charts. Then, all other charts have the scale in the middle. Very interesting, not sure what causes that yet.

ZanzyTHEbar committed:
fix: fixing requested changes

- fixed the RadiallinearScale bug
- added back in demo css styling
- removed solid-demo content
- fixed package.json

NOTE: Will add semantic-release in it's own commit

ZanzyTHEbar committed:
Format

ZanzyTHEbar committed:
feat: add styling to AllCharts

ZanzyTHEbar committed:
Format

ZanzyTHEbar committed:
fix: set App.tsx back to the default

s0ftik3 commented on solidjs-chartjs.code-workspace:
I believe it should be `solid-chartjs.code-workspace`

ZanzyTHEbar replied:
Understood.

ZanzyTHEbar committed:
fix: name of vscode workspace
